tools/ci_build/github/azure-pipelines/win-qnn-ci-pipeline.yml

```diff
@@ -62,43 +62,45 @@ jobs:
   - template: templates/jobs/win-ci-build-steps.yml
     parameters:
       WithCache: True
       Today: $(TODAY)
       AdditionalKey: "win-qnn | $(BuildConfig)"
-      BuildPyArguments: '--config $(BuildConfig) --build_dir $(Build.BinariesDirectory) --compile_no_warning_as_error --update --cmake_generator "Visual Studio 17 2022" --use_qnn --qnn_home $(QnnSDKRootDir) --parallel --use_binskim_compliant_compile_flags'
+      BuildPyArguments: >-
+        --config $(BuildConfig)
+        --build_dir $(Build.BinariesDirectory)
+        --cmake_generator "Visual Studio 17 2022"
+        --build_java
+        --use_qnn
+        --qnn_home $(QnnSDKRootDir)
+        --use_binskim_compliant_compile_flags
+        --update --parallel
       MsbuildArguments: $(MsbuildArguments)
       BuildArch: $(buildArch)
       Platform: 'x64'
       BuildConfig: $(BuildConfig)
 
-  - powershell: |
-     python $(Build.SourcesDirectory)\tools\ci_build\build.py --config $(BuildConfig) --build_dir $(Build.BinariesDirectory) --skip_submodule_sync --test --cmake_generator "Visual Studio 17 2022" --enable_onnx_tests
-    workingDirectory: '$(Build.BinariesDirectory)\$(BuildConfig)\$(BuildConfig)'
+  - script: |
+      set "ORT_JAVA_TEST_QNN_EP_BACKEND_PATH=$(QnnSDKRootDir)\lib\x86_64-windows-msvc\QnnCpu.dll"
+
+      python $(Build.SourcesDirectory)\tools\ci_build\build.py ^
+        --config $(BuildConfig) ^
+        --build_dir $(Build.BinariesDirectory) ^
+        --cmake_generator "Visual Studio 17 2022" ^
+        --build_java ^
+        --use_qnn ^
+        --qnn_home $(QnnSDKRootDir) ^
+        --use_binskim_compliant_compile_flags ^
+        --test --enable_onnx_tests
     displayName: 'Run unit tests'
 
-  # Comment out QnnCpu tests because QNN SDK 2.22 CPU backend crashes when executing MatMuls.
-  # Does not happen with HTP backend.
-  # - script: |
-  #    .\$(BuildConfig)\onnx_test_runner -j 1 -v -e qnn -i "backend_path|$(QnnSDKRootDir)\lib\x86_64-windows-msvc\QnnCpu.dll" $(Build.SourcesDirectory)\cmake\external\onnx\onnx\backend\test\data\node
-  #    workingDirectory: '$(Build.BinariesDirectory)\$(BuildConfig)'
-  #    displayName: 'Run ONNX Tests'
-  #
-  # - script: |
-  #    .\$(BuildConfig)\onnx_test_runner -j 1 -v -e qnn -i "backend_path|$(QnnSDKRootDir)\lib\x86_64-windows-msvc\QnnCpu.dll" C:\data\float32_models
-  #    workingDirectory: '$(Build.BinariesDirectory)\$(BuildConfig)'
-  #    displayName: 'Run float32 model tests'
+  - script: |
+      .\$(BuildConfig)\onnx_test_runner -j 1 -v -e qnn -i "backend_path|$(QnnSDKRootDir)\lib\x86_64-windows-msvc\QnnCpu.dll" $(Build.SourcesDirectory)\cmake\external\onnx\onnx\backend\test\data\node
+    workingDirectory: '$(Build.BinariesDirectory)\$(BuildConfig)'
+    displayName: 'Run ONNX Tests'
+
+  - script: |
+      .\$(BuildConfig)\onnx_test_runner -j 1 -v -e qnn -i "backend_path|$(QnnSDKRootDir)\lib\x86_64-windows-msvc\QnnCpu.dll" C:\data\float32_models
```

Comment:
@snnn FYI, after re-enabling this test, it's not picking up any models in `C:\data\float32_models` on the build agent.

https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1505846&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774&t=8d0a7aee-8090-5653-72a7-7c81ab87cde7&l=24

Reply:
If this isn't actually doing anything currently it would be good to capture why (and hopefully an ETA for when it should work) in a comment so that context isn't lost.

Reply:
@jywu-msft I think this image is maintained by your team. should we run these tests or not?

Reply:
discussed with @snnn offline, the image will be updated soon